specification/datafactory/resource-manager/Microsoft.DataFactory/stable/2018-06-01/entityTypes/Dataset.json

```diff
@@ -1182,6 +1182,36 @@
         }
       }
     },
+    "TeradataTableDatasetTypeProperties": {
```

Comment:
>Teradata [](start = 5, length = 8)

I am really confused about these changes. The new definitions aren't really referenced in any paths, so this will result in no change to the API contract. Autorest might generate types for all the new definitions, but the clients won't get any new methods to utilize them.

What is the intent here? If you are planning to add paths later, will you be introducing a new API version?

Reply:
I missed that we're using discriminator value.

---
In reply to: [295493306](https://github.com/Azure/azure-rest-api-specs/pull/6346#discussion_r295493306) [](ancestors = 295493306)

Reply:
The users know that they're expected to update to latest SDK to use newest types of data sets.

---
In reply to: [295580149](https://github.com/Azure/azure-rest-api-specs/pull/6346#discussion_r295580149) [](ancestors = 295580149,295493306)